service/matching/config.go

```diff
@@ -210,6 +212,8 @@ func NewConfig(
 		VisibilityDisableOrderByClause:    dc.GetBoolPropertyFnWithNamespaceFilter(dynamicconfig.VisibilityDisableOrderByClause, true),
 		VisibilityEnableManualPagination:  dc.GetBoolPropertyFnWithNamespaceFilter(dynamicconfig.VisibilityEnableManualPagination, true),
 
+		ListNexusIncomingServicesLongPollTimeout: dc.GetDurationProperty(dynamicconfig.MatchingListNexusIncomingServicesLongPollTimeout, 5*time.Minute-10*time.Second),
```

Comment:
Why are you subtracting 10 seconds? Requires a comment.

Reply:
Was copying from GetTaskQueueUserData, but probably unnecessary here.

Reply:
I think subtracting makes sense but it requires a comment, we want to make sure that the request doesn't end in timeout, better send an empty response.
And we should set the client timeout to 5 minutes when you get to that.